### PR TITLE
[WIP]support elasticsearch

### PR DESCRIPTION
**Pull request by UmmizzZ**
### Ⅰ. Describe what this PR did
初步完成了Seata对ElasticSearch的支持
### Ⅱ. Does this pull request fix one issue?

#fixes#3746

### CODE REVIEW

**Comment by lightClouds917** [comment]
Please change the title to :  [WIP]feature:support elasticsearch .
when you finish the code, remote [WIP] , we will start review the code carefully.

File: common/src/main/java/io/seata/common/util/ElasticSearchUtils.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
+package io.seata.common.util;
+
+import io.seata.common.exception.NotSupportYetException;
+import org.apache.commons.lang.ObjectUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.*;
```
Comment (lightClouds917) [inline common/src/main/java/io/seata/common/util/ElasticSearchUtils.java:10]: do not import .*, just import what you use

Comment (lightClouds917) [inline common/src/main/java/io/seata/common/util/ElasticSearchUtils.java:10]: do not import .*, just import what you use

File: server/src/main/java/io/seata/server/storage/elasticsearch/session/ElasticSearchSessionManager.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+import io.seata.common.exception.StoreException;
+import io.seata.common.executor.Initialize;
+import io.seata.common.loader.LoadLevel;
+import io.seata.common.loader.Scope;
+import io.seata.common.util.StringUtils;
+import io.seata.core.exception.TransactionException;
+import io.seata.core.model.BranchStatus;
+import io.seata.core.model.GlobalStatus;
+import io.seata.server.session.*;
```
Comment (lightClouds917) [inline server/src/main/java/io/seata/server/storage/elasticsearch/session/ElasticSearchSessionManager.java:11]: do not import *

File: core/src/main/java/io/seata/core/constants/ElasticSearchConstants.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
+package io.seata.core.constants;
+
+/**
+ * @author UmizzZ
+ * @date
+ */
+public class ElasticSearchConstants {
+    public static final String DEFAULT_GLOBAL_INDEX = "global_index";
+
+    public static final String ES_INDEX_GLOBAL_XID = "xid";
```
Comment (lightClouds917) [inline core/src/main/java/io/seata/core/constants/ElasticSearchConstants.java:10]: 同一个全局事务下，xid这种东西对global和branch都是一样的，应该不用重复定义。语义上是一致的吧。

File: server/pom.xml [hunk 4]
```diff
@@ -167,6 +167,31 @@
             <artifactId>logback-kafka-appender</artifactId>
             <version>${kafka-appender.version}</version>
         </dependency>
+
+        <!--for elasticsearch -->
+        <dependency>
+            <groupId>org.elasticsearch.client</groupId>
+            <artifactId>elasticsearch-rest-high-level-client</artifactId>
+            <version>7.13.4</version>
```
Comment (lightClouds917) [inline server/pom.xml:175]: scope应该是provided吧。我们不应该直接把版本限定死

File: server/src/main/java/io/seata/server/storage/elasticsearch/lock/ElasticSearchLocker.java [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static io.seata.core.constants.ElasticSearchConstants.*;
+
+/**
+ * @author UmizzZ
+ * @dat
```
Comment (lightClouds917) [inline server/src/main/java/io/seata/server/storage/elasticsearch/lock/ElasticSearchLocker.java:43]: @data标签都去掉吧

File: server/src/main/java/io/seata/server/storage/elasticsearch/lock/ElasticSearchLocker.java [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.index.reindex.BulkByScrollResponse;
+import org.elasticsearch.index.reindex.DeleteByQueryRequest;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
```
Comment (lightClouds917) [inline server/src/main/java/io/seata/server/storage/elasticsearch/lock/ElasticSearchLocker.java:36]: do not import *

**Comment by funky-eyes** [comment]
Due to code conflicts and age, if you are still interested in participating in the community, please resubmit this pr to the 2.x branch after resolving the conflict.
由于代码冲突以及年代久远，如果你还有兴趣参与到社区，请解决完冲突后重新请提交这份pr至2.x分支中

**Pull request closed without merging**